service_configuration_lib/spark_config.py

```diff
@@ -647,6 +647,10 @@ def _calculate_resources(
         'spark.executor.instances': str(executor_instances),
         'spark.task.cpus': str(task_cpus),
     })
+    if 'spark.cores.max' in user_spark_opts:
+        core_max = int(user_spark_opts.get('spark.cores.max', 0))
+        required_cores = executor_instances * executor_cores
```

Comment:
Looking at the code, it seems like this is the last location where executor_instances are calculated. It should be ok then.